webapp/src/utils.ts

```diff
@@ -293,21 +293,16 @@ export async function getProfilesByIds(state: GlobalState, ids: string[]): Promi
     return profiles;
 }
 
-export async function getProfilesForSessions(state: GlobalState, sessions: SessionState[]): Promise<{[sessionID: string]: UserProfile}> {
-    const ids = sessions.map((session) => session.user_id);
-    const profiles = await getProfilesByIds(state, ids);
-
-    // Returned profiles can be deduplicated so we need a map in order to
-    // produce the expected output where each session ID points to a profile.
-    const profilesMap = profiles.reduce((obj, profile) => {
-        obj[profile.id] = profile;
-        return obj;
-    }, {} as {[userID: string]: UserProfile});
-
-    return sessions.reduce((obj, session) => {
-        obj[session.session_id] = profilesMap[session.user_id];
-        return obj;
-    }, {} as {[sessionID: string]: UserProfile});
+export function getUserIDsForSessions(sessions: SessionState[]) {
+    const idsMap: {[id: string]: boolean} = {};
+    for (const session of sessions) {
+        idsMap[session.user_id] = true;
+    }
+    return Object.keys(idsMap);
+}
+
+export function getSessionsMapFromSessions(sessions: SessionState[]) {
+    return sessions.reduce((a, v) => ({...a, [v.session_id]: v}), {});
```

Comment:
nit: This is overly wasteful, no need to spread the accumulator, since you're creating as the input you can mutate it.